widgetssdk/src/main/java/com/glia/widgets/callvisualizer/CallVisualizerRepository.kt

```diff
@@ -0,0 +1,45 @@
+package com.glia.widgets.callvisualizer
+
+import com.glia.androidsdk.comms.Media
+import com.glia.androidsdk.comms.MediaDirection
+import com.glia.androidsdk.comms.MediaUpgradeOffer
+import com.glia.androidsdk.omnibrowse.Omnibrowse
+import com.glia.androidsdk.omnibrowse.OmnibrowseEngagement
+import com.glia.widgets.di.Dependencies
+import com.glia.widgets.helper.Logger
+import com.glia.widgets.helper.Utils
+import java.util.function.Consumer
+
+class CallVisualizerRepository {
+
+    companion object {
+        private val TAG = CallVisualizerRepository::class.java.simpleName
+    }
+
+    fun init() {
+        Dependencies.glia().callVisualizer.on(Omnibrowse.Events.ENGAGEMENT)
+        { engagement: OmnibrowseEngagement ->
+            Logger.d(TAG, "New Call Visualizer engagement started")
+            val upgradeOfferConsumer = prepareMediaUpgradeOfferConsumer(engagement)
+            engagement.media.on(Media.Events.MEDIA_UPGRADE_OFFER, upgradeOfferConsumer)
+        }
+        Logger.d(TAG, "CallVisualizerRepository initialized")
+    }
+
+    private fun prepareMediaUpgradeOfferConsumer(engagement: OmnibrowseEngagement): Consumer<MediaUpgradeOffer> {
+        return Consumer { offer: MediaUpgradeOffer ->
+            Logger.d(
+                TAG,
+                "upgradeOfferConsumer, offer: $offer"
+            )
+            val dialogController = Dependencies.getControllerFactory().dialogController
```

Comment:
I would like to have CallVisualizer and DialogController as an constructor parameters

Reply:
Fixed b650e5087a7e818581e83c48da3ca0462907e26e